### PR TITLE
Architecture Decision Record: Client Side Cookies

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

Introduce the Client Side Cookies Architecture Decision Record

### CODE REVIEW

File: docs/architecture/015-cookies.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Client Side Cookies
+
+## Context
+
+To read user's cookies, or not to read user cookies: that is the question. _Shakespeare_.
+
+Client side user cookies are an important part of web development architecture, but they contribute, from the server point of view, to impredictability in the rendering as seen and experienced by the user. In most cases the cookies are sent to the server as part of standard HTTP requests and the server can take care of generating the right output for the user, instad of postponing coookie driven user customization to the client side.
```
Comment (SiAdcock) [inline docs/architecture/015-cookies.md]: Couple of typos
- impredictability -> unpredictability
- instad > instead

**Comment by SiAdcock** [comment]
There's a distinction to be made regarding what cookie data ought to be read on the client side and what ought to come from the server. If frontend sends a data based on, say, the Edition cookie, this splits the cache and makes our caching less effective. It might be fine in this case as we've made the decision to do this. But if we go to another extreme, sending identity data from the server (e.g. username) instead of retrieving it on the client would split the cache intolerably.

Is it worth making this point here? Happy to discuss or clarify, and please correct me if my understanding is wrong!

**Comment by AWare** [comment]
Agree with @SiAdcock we should discuss cache partitioning here and base this on how much we're willing to split the cache per article. (And whether things like `GU_EDITION` properly qualify as cache splits as the split kind of lines up with different fastly pops/edge nodes)

File: docs/architecture/015-cookies.md (hunk 1 above)
Comment (shtukas) [reply docs/architecture/015-cookies.md]: Corrected. Thanks!

Comment (SiAdcock) [reply docs/architecture/015-cookies.md]: Thanks. I've just noticed a triple-oh `coookie` in there too 😆

**Comment by shtukas** [comment]
@SiAdcock @AWare 
I am going to close this. I am happy to come back to it when the time will be more relevant.

**Pull request closed without merging**